### ISSUE TITLE
MacOS nativeUpdater fails

### ISSUE THREAD

**Issue by ItsTonyPlayz** [FTBTeam/FTB-App-Feedback#67]
### What Operating System

MacOS (Latest)

### App Version

202110071354-4bdfd15050-release

### UI Version

4bdfd15050

### Debug Code

FTB-DBGCUHAWAKAGI

### Describe the bug

A part of the Minecraft Launcher called nativeUpdater fails to launch

### Steps to reproduce

1. Install FTB Interactions 2.0.9
2. Hit Play
3. Minecraft Launcher window opens saying "Updating Minecraft" and immediately crashes
4. Error message opens saying "nativeUpdater quits unexpectedly"

### Expected behaviour

FTB Interactions successfully launching and working properly

### Screenshots

![image](https://user-images.githubusercontent.com/84017557/138544915-01d7ed7c-0270-4dcf-8883-212f66fbb995.png)

**Comment by Raidobw2** [comment]
### Another user in the FTB App support channel on the Discord cannot use the FTB App correctly on MacOS, Monterey 12.
### Below is a report of what happened:

> I cannot launch the macOS FTBApp. When I try to, it bounces in the dock a couple of times, and then goes away. And it happens too quick for me to be able to find it in Activity Monitor.
> 
> I dug through all the Issues I can find on the github FTB-App/issues, but cannot find any that sound like they are similar.
> 
> I have also tried using App Cleaner to delete the app and then copied it from the DMG back in to Applications and tried running it again. Other than requiring me to right-click and open to get past the malware dialog, the behavior doesn't change.
> 
> What can I do from here? The FAQ in this channel doesn't point me to where the logs are on macOS, or where other directories might be that I could purge to try and get it running.

Initial state: FTB-DBGEPULAVIMOQ
User contents of `~/Library/Application Support/.ftba/` but instances, and then ran the tool again: FTB-DBGTANICEQEXU.
Tool v2 got used afterwards: FTB-DBGPEKEJAKUHO (wrong tool apparently).
Tool for MacOS from wuffs: FTB-DBGAKOPEXAYUZ mentions an "MC bin folder missing".
Another different tool used: FTB-DBGSUXAVOSUHE
User deleted `~/Library/Application Support/.ftba/`, same behavior and running FTBApp.app does not re-create that folder.
The user got given another dmg to install the FTB App, same behavior, didn't work.
User checked: Go to settings -> security & privacy -> See if there is a button asking you to allow an app.
They are opening the FTB App correctly with the right permissions.

Some behavior requested by devs:
electron.log:
```
[2021-11-07 13:05:12.422] [info]  We have a --ws
[2021-11-07 13:05:12.428] [debug] Got the lock
[2021-11-07 13:05:12.504] [warn]  (electron) The default value of app.allowRendererProcessReuse is deprecated, it is currently "false".  It will change to be "true" in Electron 9.  For more information please check https://github.com/electron/electron/issues/18397
[2021-11-07 13:05:12.529] [error] (node:3425) ProtocolDeprecateCallback: The callback argument of protocol module APIs is no longer needed.
```

Running `/Applications/FTBApp.app/Contents/Resources/app/launcher-202105051818-5b1f30277f-release-all.jar` shows this, finally:
![image](https://user-images.githubusercontent.com/17010632/140673013-18fe4609-c978-4663-9aa9-bdfac2b82f3f.png)

User ran `FTBA_macos_202111051727-fc8797cf31-release.dmg`.

> After upgrading to the latest, and launching it per the right-click Open x2 process I mentioned above, the file doesn't exist:
```
$ ls -a /Applications/FTBApp.app/Contents/Resources/app/bin/ftbapp.app/Contents/MacOS/electron.log
ls: /Applications/FTBApp.app/Contents/Resources/app/bin/ftbapp.app/Contents/MacOS/electron.log: No such file or directory
```

> Running `/Applications/FTBApp.app/Contents/Resources/app/launcher-202111051727-fc8797cf31-release-all.jar`  directly creates that log file, and also opens a window that looks right?

![image](https://user-images.githubusercontent.com/17010632/140673121-b9913d0e-0c0b-4b5d-bf7a-f2301c5dcc55.png)
Logs at that moment:
```
$ find /Applications/FTBApp.app/ -iname '*.log' -print   
/Applications/FTBApp.app//Contents/Resources/app/bin/ftbapp.app/Contents/MacOS/electron.log
```

```
$ cat /Applications/FTBApp.app//Contents/Resources/app/bin/ftbapp.app/Contents/MacOS/electron.log
[2021-11-07 13:13:19.828] [info]  We have a --ws
[2021-11-07 13:13:19.834] [debug] Got the lock
[2021-11-07 13:13:20.969] [error] (node:3754) ExtensionLoadWarning: Warnings loading extension at /Users/charles/Library/Application Support/ftb-app/extensions/nhdogjmejiglipccpnnnanhbledajbpd:
  Unrecognized manifest key 'browser_action'.
  Unrecognized manifest key 'update_url'.
  Permission 'contextMenus' is unknown or URL pattern is malformed.
  Cannot load extension with file or directory name _metadata. Filenames starting with "_" are reserved for use by the system.

(Use `ftbapp --trace-warnings ...` to show where the warning was created)
```

```
$/Applications/FTBApp.app/Contents/Resources/jre.bundle/Contents/Home/bin/java --version
openjdk 11.0.6 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.6+10)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.18.1, JRE 11 Mac OS X amd64-64-Bit 20200122_411 (JIT enabled, AOT enabled)
OpenJ9   - 51a5857d2
OMR      - 7a1b0239a
JCL      - da35e0c380 based on jdk-11.0.6+10)
```

> For fun, I tried running `/Applications/FTBApp.app/Contents/Resources/app/bin/ftbapp.app` and got this crash report: https://pste.ch/ufelicikix.sql

Workaround by our devs, running this to start the app:
`/Applications/FTBApp.app/Contents/Resources/jre.bundle/Contents/Home/bin/java -jar /Applications/FTBApp.app/Contents/Resources/app/launcher-202105051818-5b1f30277f-release-all.jar`

User tried that last workaround:
```
$ java --version                           
java 16.0.1 2021-04-20
Java(TM) SE Runtime Environment (build 16.0.1+9-24)
Java HotSpot(TM) 64-Bit Server VM (build 16.0.1+9-24, mixed mode, sharing)
```

> Although, that specific command you gave me produced an interesting result: https://pste.ch/lovakiwoxo.cs

!!!! Dev mentions:

> Me thinks that version of java has an issue with the beta version of macos

TL;DR: Java 16 crashes the workaround and Java 11 works.
~End of discussion on the Discord.

**Comment by sethtaplin** [comment]
Running Monterey 12.0.1 the equivalent workaround command
`/Applications/FTBApp.app/Contents/Resources/jre.bundle/Contents/Home/bin/java -jar /Applications/FTBApp.app/Contents/Resources/app/launcher-202111051727-fc8797cf31-release-all.jar`
results in a JNI error:
`Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.NoClassDefFoundError: java.util.zip.Inflater (initialization failure)
	at java.base/java.lang.J9VMInternals.initializationAlreadyFailed(J9VMInternals.java:143)
	at java.base/java.util.zip.ZipFile$CleanableResource.getInflater(ZipFile.java:745)
	at java.base/java.util.zip.ZipFile$ZipFileInflaterInputStream.<init>(ZipFile.java:439)
	at java.base/java.util.zip.ZipFile.getInputStream(ZipFile.java:406)
	at java.base/java.util.jar.JarFile.getBytes(JarFile.java:798)
	at java.base/java.util.jar.JarFile.getManifestFromReference(JarFile.java:419)
	at java.base/java.util.jar.JarFile.getManifest(JarFile.java:406)
	at java.base/sun.launcher.LauncherHelper.getMainClassFromJar(LauncherHelper.java:522)
	at java.base/sun.launcher.LauncherHelper.loadMainClass(LauncherHelper.java:747)
	at java.base/sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:655)
Caused by: java.lang.UnsatisfiedLinkError: java/util/zip/Inflater.initIDs()V
	at java.base/java.util.zip.Inflater.<clinit>(Inflater.java:123)
	at java.base/java.util.zip.ZipFile$CleanableResource.getInflater(ZipFile.java:745)
	at java.base/java.util.zip.ZipFile$ZipFileInflaterInputStream.<init>(ZipFile.java:439)
	at java.base/java.util.zip.ZipFile.getInputStream(ZipFile.java:406)
	at java.base/java.util.jar.JarFile.getBytes(JarFile.java:798)
	at java.base/java.util.jar.JarFile.checkForSpecialAttributes(JarFile.java:1005)
	at java.base/java.util.jar.JarFile.isMultiRelease(JarFile.java:388)
	at java.base/java.util.jar.JarFile.getEntry(JarFile.java:507)
	at java.base/java.util.jar.JarFile.getJarEntry(JarFile.java:468)
	at java.base/jdk.internal.util.jar.JarIndex.getJarIndex(JarIndex.java:115)
	at java.base/jdk.internal.loader.URLClassPath$JarLoader$1.run(URLClassPath.java:876)
	at java.base/jdk.internal.loader.URLClassPath$JarLoader$1.run(URLClassPath.java:864)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:770)
	at java.base/jdk.internal.loader.URLClassPath$JarLoader.ensureOpen(URLClassPath.java:863)
	at java.base/jdk.internal.loader.URLClassPath$JarLoader.<init>(URLClassPath.java:838)
	at java.base/jdk.internal.loader.URLClassPath$3.run(URLClassPath.java:604)
	at java.base/jdk.internal.loader.URLClassPath$3.run(URLClassPath.java:587)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:770)
	at java.base/jdk.internal.loader.URLClassPath.getLoader(URLClassPath.java:586)
	at java.base/jdk.internal.loader.URLClassPath.getLoader(URLClassPath.java:550)
	at java.base/jdk.internal.loader.URLClassPath.getResource(URLClassPath.java:413)
	at java.base/jdk.internal.loader.URLClassPath.getResource(URLClassPath.java:469)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:895)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:806)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:764)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:1078)`
	
However, running with an Oracle java 11 installation did work.
`java -version
java version "11.0.13" 2021-10-19 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.13+10-LTS-370)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.13+10-LTS-370, mixed mode)`

**Comment by MichaelHillcox** [comment]
Added to main list of launch issues #119 thanks for your help by reporting your issue, we're looking into it :+1:
